patroni/utils.py

```diff
@@ -416,3 +418,27 @@ def cluster_as_json(cluster):
         if cluster.failover.candidate:
             ret['scheduled_switchover']['to'] = cluster.failover.candidate
     return ret
+
+
+def merge_paths(d1, d2):
+    real_d1 = os.path.realpath(d1)
+    real_d2 = os.path.realpath(os.path.join(real_d1, d2))
+    return (real_d2, (os.path.commonprefix([real_d1+os.path.sep, real_d2+os.path.sep]) == real_d1+os.path.sep))
+
+
+def validate_directory(d, msg="{} {}"):
+    try:
+        if not os.path.exists(d):
+            os.makedirs(d)
+        elif os.path.isdir(d):
+            try:
+                fd, tmpfile = tempfile.mkstemp(dir=d)
+                os.close(fd)
+                os.remove(tmpfile)
+            except OSError:
+                raise OSError("{} is not writable.".format(d))
```

Comment:
Why not just immediately raise the `PatroniException` with the correctly formatted message and without the directory name redundancy?